docs: expand docs for SMS recovery (#2261)
ory/docs: docs/identities/get-started/account-recovery.mdx

### Enable SMS account recovery

> This feature is enabled via a project-level feature flag, which is `on` by default for new projects in Ory Network. projects.

Reviewer: Looks like an errant period "Ory Network. projects."

This feature only works when the recovery strategy is `code` (not `link`).

#### As a user

> For users with only an email address and no phone number as identifier, the recovery process is unchanged.

Reviewer: "For users that use only email as their identifier (a phone number was not provided), the recovery process remains the same."

> To receive a recovery code via SMS, the users phone number must be registered as a recovery address.

Reviewer: Add apostrophe "the user's phone number"

If the user has multiple recovery addresses registered, they will be prompted to choose one from a masked list during the recovery
process.

To confirm ownership, the user may be asked to enter the complete address after selecting it. This verification step is skipped if
they already entered that same address to begin the recovery process.

#### As a developer

> Enable the feature flag `choose_recovery_address` to be able to send a recovery code via SMS.

Reviewer: switch order of 'feature flag' and `choose_recovery_address` 
remove 'be able to'
remove 'via'
"Enable the `choose_recovery_address` feature flag to allow sending a recovery code using SMS."
Question: Why refer to the feature flag (I presume this is the flag name in code) if the instructions are to use the console UI? Do they need to know it for using the CLI? Are we co-mingling the Console and UI instructions here?

### Enable SMS account recovery

> This is enabled by turning a feature flag to 'on' at the project level (opt-in). This flag is enabled by default for newly created

Reviewer 2: ```suggestion
This feature is enabled via a project-level feature flag, which is `on` by default for new projects in Ory Network.
```

can you please also provide the link to where you enable this in the console for existing projects?

Author: > can you please also provide the link to where you enable this in the console for existing projects?

this is present a few line below

projects.

This feature only works when the recovery strategy is `code` (not `link`).

#### As a user

> If you only have an email address and no phone number, there is no perceivable difference.

Reviewer 2: ```suggestion
For users with only an email address and no phone number as identifier, the recovery process is unchanged.
```

> If you do have a phone number and want to receive a SMS containing the recovery code, make sure that this phone number is
> registered as a recovery address.

Reviewer 2: ```suggestion
To receive a recovery code via SMS, the users phone number must be registered as a recovery address.
```

> If you have multiple recovery addresses registered, during the recovery flow, you will be presented a list of your (masked)
> addresses and you will be prompted to choose one of them.

Reviewer 2: ```suggestion
If the user has multiple recovery addresses registered, they will be prompted to choose one from a masked list during the recovery process.
```

> You may also be prompted to enter this address in full to prove that you know the complete address, for security reasons.

Reviewer 2: what means "may" here? is the user prompted to provide the full address or not @gaultier

Reviewer 2: So basically if the user inputs their address in step 1 and then chooses that address in step 3 they wont be asked to input the full address bc they did already, that makes sense. 

how about 
```suggestion
To confirm ownership, the user may be asked to enter the complete address after selecting it. This verification step is skipped if they already entered that same address to begin the recovery process.
```

> #### As an Administrator

Reviewer 2: what is the difference between developer and administrator here @gaultier ?

> #### As an Administrator

Reviewer 2: Yea I think we can have it in one section for now as most Ory Network users are both admin and dev in the same person 😁

4. Toggle "Receive a recovery code via SMS" to 'on'
5. Click **Save**

> It is safe to toggle this feature flag. Changes only take effect when a new recovery flow is initiated; existing flows are

Reviewer: Do you mean when a new recovery flow is added (by a developer) to the system or when a user initiates a new recovery flow?

Author: The latter, the former is not a thing in our system. Adding a new flow can only be done by us by writing code.

unaffected.

A [SMS HTTP gateway](../../kratos/emails-sms/10_sending-sms.mdx) must be configured to deliver SMS messages.

##### API changes

> When using the built-in Account Experience UI, there is nothing special to do. Otherwise, you need to do some changes.

Reviewer: Do you mean that if the developer/admin changes the recovery method in the console UI, then they do not need to make any API changes? 
Is this section an alternate (API) method to change the recovery method?

Author: Good points all around, let's split the explanations by type of product (EOL/OSS, Network). I think it makes more sense.

New fields have been added to the payload when getting and updating the recovery flow that are exclusively used when this feature
flag is enabled:

- `recovery_address`
- `recovery_select_address`
- `recovery_confirm_address`
- `screen`

And these fields will have no effect anymore (they can be present in the payload, but will be ignored):

- `email`

When integrating with the API directly, your applications must send and receive these fields. Refer to the
[API documentation](../../reference/api#tag/frontend/operation/updateRecoveryFlow) for more details.

> A maximum of 10 recovery addresses will be shown to a user during the recovery flow.

Reviewer: If this is possible instead of certain, change 'will' to 'can'
"A maximum of 10 recovery addresses can be shown to a user during the recovery flow."

Author: Hmm not sure... What I want to express is: There is a hard-coded limit in the system so that up to 10 recovery addresses are retrieved from the database and returned to the browser.

4. Toggle "Receive a recovery code via SMS" to 'on'
5. Click **Save**

It is safe to toggle this feature flag on and off. This flag only is take into account when a new recovery flow is created.
Existing flows will be unaffected by toggling the flag.

SMSes can only be delivered if a [SMS HTTP gateway](../../kratos/emails-sms/10_sending-sms.mdx) is configured.

#### As a developer

> When using `elements`, there is nothing special to do. Otherwise, you need to do some changes.

Author: I think I meant here Account Experience, not elements. Will fix.
